hackathon_site/dashboard/frontend/src/api/helpers.ts

```diff
@@ -0,0 +1,79 @@
+import {
+    Order,
+    OrderInTable,
+    OrderItemTableRow,
+    ReturnedItem,
+    ReturnOrderInTable,
+} from "./types";
+
+export const teamOrderListSerialization = (
+    orders: Order[]
+): {
+    pendingOrders: OrderInTable[];
+    checkedOutOrders: OrderInTable[];
+    returnedOrders: ReturnOrderInTable[];
+    hardwareIdsToFetch: number[];
+} => {
+    const pendingOrders: OrderInTable[] = [];
+    const checkedOutOrders: OrderInTable[] = [];
+    const returnedOrders: ReturnOrderInTable[] = [];
+    const hardwareIdsToFetch: Record<number, number> = {};
+    orders.forEach((order) => {
+        if (order.status !== "Cancelled") {
+            const hardwareItems: Record<number, OrderItemTableRow> = {};
+            const hardwareRequested: Record<number, number> = {};
```

Comment:
What's the advantage of using Record type in these cases?

Reply:
Record is like a generic javascript object where the first argument is the key type and the second is the value type, e.g. Record<type1, type2> creates the type
`{ [key: type1]: type2 }`

Reply:
you can learn more from typescript documentation here: https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type

Reply:
@Leo6Leo it's not creating a dict it's just **defining the type** of an object (which is like a dict in python)